core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -462,6 +481,34 @@ private void tryStart(Instant startedAt) {
         }
     }
 
+    @VisibleForTesting
+    Checksum hashCopiedFiles() {
+        Checksum checksum = new Adler32();
+        copyToFileContainerPathMap.entrySet().stream().sorted(Entry.comparingByValue()).forEach(entry -> {
+            byte[] pathBytes = entry.getValue().getBytes();
+            // Add path to the hash
+            checksum.update(pathBytes, 0, pathBytes.length);
+
+            File file = new File(entry.getKey().getResolvedPath());
+            checksumFile(file, checksum);
+        });
+        return checksum;
+    }
+
+    @VisibleForTesting
+    @SneakyThrows(IOException.class)
+    void checksumFile(File file, Checksum checksum) {
```

Comment:
It doesn't look like the permission/mode for files or directories gets included in the checksum, but it probably should be.

[`FileUtils.checksum`](https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/FileUtils.html#checksum(java.io.File,%20java.util.zip.Checksum)) only seems to operate on the content of the file, not FS permissions - is that right?

Reply:
unlike volumes, copy does not preserve the file mode.

We may want to add it in future (e.g. by `exec`-ing after the copy), I guess we can start hashing it then too

Reply:
I was wrong, we do have the permissions. Fixing it...

Reply:
Fixed! (some time ago, actually, but the push didn't work last time :D)